Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/impl/apmlog/LogFilterControllerWithAPMLog.java

```diff
@@ -24,55 +24,193 @@
 import org.apromore.plugin.portal.processdiscoverer.data.InvalidDataException;
+import org.eclipse.collections.impl.map.mutable.UnifiedMap;
+import org.zkoss.json.JSONObject;
 import org.zkoss.zk.ui.Session;
 import org.zkoss.zk.ui.Sessions;
 import org.zkoss.zk.ui.event.Event;
 import org.zkoss.zk.ui.event.EventListener;
 import org.zkoss.zk.ui.event.EventQueue;
 import org.zkoss.zk.ui.event.EventQueues;
+import org.zkoss.zk.ui.util.Clients;
+import org.zkoss.zul.Messagebox;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
 /**
  * LogFilterControllerWithAPMLog is {@link LogFilterController} but uses APMLog to do filtering.
- * 
+ *
  * @author Bruce Nguyen
  *
  */
-public class LogFilterControllerWithAPMLog extends LogFilterController implements LogFilterResultListener {
+public class LogFilterControllerWithAPMLog extends LogFilterController implements LogFilterClient {
     private LogDataWithAPMLog logData;
     public LogFilterControllerWithAPMLog(PDController controller) throws Exception {
         super(controller);
         if (!(parent.getLogData() instanceof LogDataWithAPMLog)) {
             throw new InvalidDataException("Expect LogDataWithAPMLog data but receiving different data!");
         }
         else {
-            logData = (LogDataWithAPMLog)controller.getLogData();            
+            logData = (LogDataWithAPMLog)controller.getLogData();
         }
     }
 
     @Override
     // Open LogFilter window
     public void onEvent(Event event) throws Exception {
-        // Store in session for LogFilterEE to work with APMLog
-        Session session = Sessions.getCurrent();
-        session.setAttribute("apmlog_tobe_filtered", logData.getOriginalAPMLog());
-        session.setAttribute("filtered_log_name", parent.getTitle()); // required for showing log name on the title of FilterEE
-        Sessions.getCurrent().setAttribute("sourceLogId", parent.getSourceLogId());
-
-        parent.getLogFilterPlugin().execute(new LogFilterContext(parent.getContextData().getPortalContext()), 
-                new LogFilterInputParams(
-                        parent.getLogData().getLog(), 
-                        parent.getUserOptions().getMainAttributeKey(), 
-                        parent.getLogData().getCurrentFilterCriteria()),
-                        this);
-
-        subscribeFilterResult();
+        if (event.getData() == null) {
+            LogFilterRequest lfr = logData.getCurrentFilterCriteria() == null ||
+                    ((List<LogFilterRule>) logData.getCurrentFilterCriteria()).isEmpty() ?
+                    getRequestWithOption(new EditorOption(FilterType.CASE_VARIANT)) : getDefaultRequest();
+            parent.getLogFilterPlugin().execute(lfr);
+        } else {
+            if (event.getData() instanceof JSONObject) onInvokeExtEvent((JSONObject) event.getData());
+            else onInvokeEvent(event.getData().toString());
+        }
+    }
```

Comment:
It'd be a little easier to see the three possibilities here if the conditional were all at the same level of indentation.  Something like:
```
if (event.getData() == null) {
    ....
} else if (event.getData() instanceof JSONObject) {
    ....
} else {
    ....
}
```